include/kitepp/responses/market.hpp

```diff
@@ -0,0 +1,359 @@
+            for (auto& i : sellDepthVal.GetArray()) {
+                sell.emplace_back(i.GetObject());
+            };
+        }
+
+        std::vector<depth> buy;
+        std::vector<depth> sell;
+    } marketDepth;
+};
+
+/// Represents a ohlc quote information of an instrument.
+struct ohlcQuote {
+    ohlcQuote() = default;
+    explicit ohlcQuote(const rj::Value::Object& val) { parse(val); };
+
+    void parse(const rj::Value::Object& val) {
+        instrumentToken = utils::json::get<int>(val, "instrument_token");
+        lastPrice = utils::json::get<double>(val, "last_price");
+        OHLC = utils::json::get<utils::json::JsonObject, ohlc>(val, "ohlc");
+    };
+
+    int instrumentToken = -1;
+    double lastPrice = -1;
+    ohlc OHLC;
+};
+
+/// Represents a LTP quote information of an instrument.
+struct ltpQuote {
+    ltpQuote() = default;
+    explicit ltpQuote(const rj::Value::Object& val) { parse(val); };
+
+    void parse(const rj::Value::Object& val) {
+        instrumentToken = utils::json::get<int>(val, "instrument_token");
+        lastPrice = utils::json::get<double>(val, "last_price");
+    };
+
+    int instrumentToken = -1;
+    double lastPrice = -1;
+};
+
+/// represents parameters required for the `getHistoricalData` function
+struct historicalDataParams {
+    GENERATE_FLUENT_METHOD(
+        historicalDataParams, int, instrumentToken, InstrumentToken);
+    GENERATE_FLUENT_METHOD(historicalDataParams, bool, continuous, Continuous);
+    GENERATE_FLUENT_METHOD(historicalDataParams, bool, oi, Oi);
+    GENERATE_FLUENT_METHOD(historicalDataParams, const string&, from, From);
+    GENERATE_FLUENT_METHOD(historicalDataParams, const string&, to, To);
+    GENERATE_FLUENT_METHOD(
+        historicalDataParams, const string&, interval, Interval);
+
+    int instrumentToken = -1;
+    bool continuous = false;
+    bool oi = false;
+    string from;
+    string to;
+    string interval;
+};
+
+/// Represents historical data of an instrument.
+struct historicalData {
+    historicalData() = default;
+    explicit historicalData(const rj::Value::Array& val) { parse(val); };
+
+    void parse(const rj::Value::Array& val) {
+        // if the sent value doesn't have a floating point (this time),
+        // GetDouble() will throw error
+        static auto getDouble = [](rj::Value& val) -> double {
+            if (val.IsDouble()) { return val.GetDouble(); };
+            if (val.IsInt()) { return val.GetInt(); };
+            throw libException("type isn't double");
+        };
+        datetime = val[DATETIME_IDX].GetString();
+        open = getDouble(val[OPEN_IDX]);
+        high = getDouble(val[HIGH_IDX]);
+        low = getDouble(val[LOW_IDX]);
+        close = getDouble(val[CLOSE_IDX]);
+        volume = val[VOLUME_IDX].GetInt();
+        if (val.Size() > OI_IDX) { OI = val[OI_IDX].GetInt(); };
+    };
+
+    int volume = -1;
+    int OI = -1;
```

Comment:
Correct me if I'm wrong, `int` is 32 bit by default, irrespective of arch, right? Then these types should be changed to 64 bit. Better to change all instances of Volume and OI fields. This recently caused overflow issues in .Net lib. Ref: https://github.com/zerodha/dotnetkiteconnect/issues/34

Reply:
Will do.

Might be a good idea to mention precise types in the API docs too. Right now, [quotes section has volume type as just `int`](https://kite.trade/docs/connect/v3/market-quotes/#response-attributes). This led me to believe that volume is `int32` because `instrument_token` does have precise type mentioned (`uint32`).